code/game/objects/noose.dm

```diff
@@ -0,0 +1,153 @@
+			M.Knockdown(60)
+		if(istype(M, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			H.noosed = FALSE
+		unbuckle_all_mobs(force=1)
+		M.pixel_z = initial(M.pixel_z)
+		pixel_z = initial(pixel_z)
+		add_fingerprint(user)
+
+/obj/structure/chair/noose/user_buckle_mob(mob/living/carbon/human/M, mob/user)
+	if(!in_range(user, src) || user.stat || user.restrained() || !iscarbon(M))
+		return FALSE
+
+	if (!M.get_bodypart("head"))
+		to_chat(user, "<span class='warning'>[M] has no head!</span>")
+		return FALSE
+
+	if(M.loc != src.loc)
+		return FALSE //Can only noose someone if they're on the same tile as noose
+
+	add_fingerprint(user)
+
+	M.visible_message(\
+		"<span class='danger'>[user] attempts to tie \the [src] over [M]'s neck!</span>",\
+		"<span class='userdanger'>[user] ties \the [src] over your neck!</span>")
+	if(user != M)
+		to_chat(user, "<span class='notice'>It will take 20 seconds and you have to stand still.</span>")
+	if(do_mob(user, M, user == M ? 0:200))
+		if(buckle_mob(M))
+			user.visible_message(\
+				"<span class='warning'>[user] ties \the [M != user ? "[user]" : "their"][src] over [M]'s neck!</span>",\
+				"<span class='userdanger'>[M != user ? "[user] ties" : "You tie"] \the [src] over your neck!</span>")
+			playsound(user.loc, 'sound/effects/noosed.ogg', 50, 1, -1)
+			log_combat(user, M, "hanged", src)
+			M.noosed = TRUE
+			return TRUE
+	user.visible_message(\
+		"<span class='warning'>[user] fails to tie \the [src] over [M]'s neck!</span>",\
+		"<span class='warning'>You fail to tie \the [src] over [M]'s neck!</span>")
+	return FALSE
+
+
+/obj/structure/chair/noose/process()
+	if(!has_buckled_mobs())
+		STOP_PROCESSING(SSobj, src)
+		return
+	for(var/m in buckled_mobs)
+		var/mob/living/buckled_mob = m
+		if(pixel_x >= 0)
+			animate(src, pixel_x = -3, time = 45, easing = ELASTIC_EASING)
+			animate(m, pixel_x = -3, time = 45, easing = ELASTIC_EASING)
+		else
+			animate(src, pixel_x = 3, time = 45, easing = ELASTIC_EASING)
+			animate(m, pixel_x = 3, time = 45, easing = ELASTIC_EASING)
+		if(buckled_mob.mob_has_gravity())
+			buckled_mob.adjustOxyLoss(5)
+			if(prob(40))
+				buckled_mob.emote("gasp")
+			if(prob(20))
+				var/flavor_text = list("<span class='suicide'>[buckled_mob]'s legs flail for anything to stand on.</span>",\
+										"<span class='suicide'>[buckled_mob]'s hands are desperately clutching the noose.</span>",\
+										"<span class='suicide'>[buckled_mob]'s limbs sway back and forth with diminishing strength.</span>")
+				if(buckled_mob.stat == DEAD)
+					flavor_text = list("<span class='suicide'>[buckled_mob]'s limbs lifelessly sway back and forth.</span>",\
+										"<span class='suicide'>[buckled_mob]'s eyes stare straight ahead.</span>")
+				buckled_mob.visible_message(pick(flavor_text))
+				playsound(buckled_mob.loc, 'sound/effects/noose_idle.ogg', 30, 1, -3)
+
+/mob/living/carbon/human
+	var/noosed = FALSE
```

Comment:
No. We don't need a `human` var to see if someone is being hanged. Just check if they're in the structure's `buckled_mobs` or not to keep track of this.

Reply:
so, change var/noosed to a buckled mobs check instead, got it

Reply:
>removed this and the `checknoosedrop()`proc itself as its not needed it seems

Reply:
also updated code to have buckled mob check instead of ifnoosed